draft-ietf-wimse-s2s-protocol.md

```diff
@@ -313,17 +313,14 @@ A WPT contains the following:
     * `exp`: The expiration time of the WIT (as defined in {{Section 4.1.4 of RFC7519}}). WPT lifetimes MUST be short,
      e.g., on the order of minutes or seconds.
     * `jti`: A unique identifier for the token.
+
+Optionally, the `Workload Proof Token` (WPT) may include the following, additional claims, that restrict the context further:
     * `ath`: Hash of the OAuth access token, if present in the request, which might convey end-user identity and
      authorization context of the request. The value, as per {{Section 4.1 of RFC9449}},
      is the base64url encoding of the SHA-256 hash of the ASCII encoding of the access token's value.
-    * `tth`: Hash of the Txn-Token {{?I-D.ietf-oauth-transaction-tokens}}, if present in the request,
-     which might convey end-user identity and authorization context of the request. The value MUST be the result of
-     a base64url encoding (as defined in {{Section 2 of RFC7515}}) of the SHA-256 hash of
-     the ASCII encoding of the associated token's value.
-    * `oth`: Hash of any other token in the request that might convey end-user identity and authorization context of the
-     request. The value MUST be the result of a base64url encoding (as defined in {{Section 2 of RFC7515}}) of the
-     SHA-256 hash of the ASCII encoding of the associated token's value.
-     (note: this is less than ideal but seems we need something like this for extensibility)
+    * `signed_headers`: A key-value map of headers where
+         * key: The name of the header field. In HTTP this can be generic, request or entity fields. This may differ in other protocols.
```

Comment:
My thinking here is, that not restricting it to HTTP allows `WPT` to be used for example with `Kafka` or `gRPC headers`.

Reply:
Yes and no. The normative text should be focused on HTTP. If we add a note about a future non-HTTP variant of this protocol, it should be very clear that it's outside the normative "flow".

Reply:
I'm not sure I know what exactly "generic, request or entity fields" means exactly in HTTP but I'm pretty sure it's not what should be here (especially entity).

Reply:
I'm in agreement that we want to focus the POP mechanism on HTTP. Trying to anticipate some other protocol or constraints at this point is difficult. We will need to define the POP for other protocols in other documents.  I think there is value in having the POP be flexible.  I think that is an argument in favor of HTTPSigs.  
Just to mention here It would be great if we keep WIT as not tightly bound to HTTP as it is now.